docs/sts-migrate-to-private-bucket.md

```diff
@@ -0,0 +1,491 @@
+    },
+    {
+      "Sid": "Stmt1678336953197",
+      "Action": [
+        "s3:GetObject",
+        "s3:ListBucket",
+        "s3:PutObject"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+```
+
+
+## Validate and Backup <a name="validate-backup"></a>
+
+### Validate tokens <a name="validate-token"></a>
+
+This section is to make sure everything is working correctly in your existing environment.
+
+The steps described below will use the credentials provided to the `machine-api` component, trying to assume the role using `aws-cli`.
+
+It's expected that the existing token will be able to authenticate in AWS. Otherwise, you must abort the operations and do not try to run any step described in the next sections
+
+```bash
+## test existing token
+# Get Token path from AWS credentials mounted to pod
+TOKEN_PATH=$(oc get secrets aws-cloud-credentials \
+    -n openshift-machine-api \
+    -o jsonpath='{.data.credentials}' |\
+    base64 -d |\
+    grep "^web_identity_token_file" |\
+    awk '{print$3}')
+
+# Get Controler's pod
+CAPI_POD=$(oc get pods -n openshift-machine-api \
+    -l api=clusterapi \
+    -o jsonpath='{.items[*].metadata.name}')
+
+# Extract tokens from the pod
+TOKEN=$(oc exec -n openshift-machine-api ${CAPI_POD} \
+    -c machine-controller -- cat ${TOKEN_PATH})
+
+echo $TOKEN | awk -F. '{ print $2 }' | base64 -d 2>/dev/null | jq .iss
+
+IAM_ROLE=$(oc get secrets aws-cloud-credentials \
+    -n openshift-machine-api \
+    -o jsonpath='{.data.credentials}' |\
+    base64 -d |\
+    grep "^role_arn" |\
+    awk '{print$3}')
+
+echo $IAM_ROLE
+
+aws sts assume-role-with-web-identity \
+    --role-arn "${IAM_ROLE}" \
+    --role-session-name "my-session" \
+    --web-identity-token "${TOKEN}"
+```
+
+### Backup existing state <a name="backup"></a>
+
+- Get Objects and existing authentication response
+
+```bash
+export BACKUP_PATH=$PWD/current-cluster
+mkdir $BACKUP_PATH
+
+oc get authentication -o yaml |tee -a $BACKUP_PATH/authentication.yaml
+
+aws sts assume-role-with-web-identity \
+    --role-arn "${IAM_ROLE}" \
+    --role-session-name "my-session" \
+    --web-identity-token "${TOKEN}" \
+    | jq -r '.Credentials=""' \
+    | tee ${BACKUP_PATH}/identities.json
+```
+
+- Save the current IAM Roles
+
+```bash
+aws iam list-roles \
+    | jq -r  ".Roles[] | select(.RoleName | startswith(\"${CLUSTER_NAME}-openshift\"))" \
+    | tee ${BACKUP_PATH}/iam-roles.json
+```
+
+## Migrate
+
+### Setup new OIDC with CloudFront Distribution <a name="setup"></a>
+
+#### Create the CloudFront Distribution <a name="setup-cloudfront"></a>
+
+- Create the Origin Access Identity (OAI)
+
+```bash
+export DIR_CCO="./"
+export OIDC_BUCKET_PATH="/pvt"
+
+export OAI_CLOUDFRONT_ID=$(aws cloudfront create-cloud-front-origin-access-identity \
+    --cloud-front-origin-access-identity-config \
+    CallerReference="${OIDC_BUCKET_NAME}",Comment="OAI-${OIDC_BUCKET_NAME}" \
+    | jq -r .CloudFrontOriginAccessIdentity.Id)
```

Comment:
I created a new user jshu-sts with inline policy + permission json listed at beginning.
It reported the following error.
[cloud-user@preserve-for-hive-test testdir]$ export OAI_CLOUDFRONT_ID=$(aws cloudfront create-cloud-front-origin-access-identity \
>     --cloud-front-origin-access-identity-config \
>     CallerReference="${OIDC_BUCKET_NAME}",Comment="OAI-${OIDC_BUCKET_NAME}" \
>     | jq -r .CloudFrontOriginAccessIdentity.Id)

An error occurred (AccessDenied) when calling the CreateCloudFrontOriginAccessIdentity operation: User: arn:aws:iam::301721915996:user/jshu-sts is not authorized to perform: cloudfront:CreateCloudFrontOriginAccessIdentity on resource: arn:aws:cloudfront::301721915996:origin-access-identity/* because no identity-based policy allows the cloudfront:CreateCloudFrontOriginAccessIdentity action

When I switched back my admin account, then it worked. Maybe the some required permission is missing.